core/src/main/java/org/infinispan/functional/impl/MapFactory.java

```diff
@@ -0,0 +1,33 @@
+package org.infinispan.functional.impl;
+
+import org.infinispan.functional.FunctionalMap;
+
+/**
+ * Created by katiaaresti on 29/06/17.
+ */
+public class MapFactory {
```

Comment:
Hmm, this means that encoding won't work with existing code... I would rather see ReadOnlyMapImpl et all aware of Encoded* classes, maybe even get the encoder functionality directly in the ReadOnlyMapImpl et all. There's huge code duplication anyway.